src/proposals/proposalTypes/HybridProposal.sol

```diff
@@ -551,8 +624,105 @@ abstract contract HybridProposal is
         );
 
         {
+            address wormholeCoreBase = addresses.getAddress(
+                "WORMHOLE_CORE_BASE",
+                baseChainId
+            );
+            address wormholeCoreBaseSepolia = addresses.getAddress(
+                "WORMHOLE_CORE_SEPOLIA_BASE",
+                baseSepoliaChainId
+            );
+            address temporalGov = addresses.getAddress(
+                "TEMPORAL_GOVERNOR",
+                sendingChainIdToReceivingChainId[block.chainid]
+            );
+            address wormholeCoreMoonbeam = block.chainid == moonBeamChainId
+                ? addresses.getAddress(
+                    "WORMHOLE_CORE_MOONBEAM",
+                    moonBeamChainId
+                )
+                : addresses.getAddress(
+                    "WORMHOLE_CORE_MOONBASE",
+                    moonBaseChainId
+                );
+
+            address[] memory targets = new address[](baseActions.length);
+            uint256[] memory values = new uint256[](baseActions.length);
+            bytes[] memory calldatas = new bytes[](baseActions.length);
+
+            for (uint256 i = 0; i < baseActions.length; i++) {
+                targets[i] = baseActions[i].target;
+                values[i] = baseActions[i].value;
+                calldatas[i] = baseActions[i].data;
+            }
+
+            /// assert wormhole core BASE address is not in the list of targets on Moonbeam
+            for (uint256 i = 0; i < targets.length; i++) {
+                require(
+                    targets[i] != wormholeCoreBase,
+                    "Wormhole Core BASE address should not be in the list of targets"
+                );
+                require(
+                    targets[i] != wormholeCoreBaseSepolia,
+                    "Wormhole Core BASE Sepolia address should not be in the list of targets"
+                );
+            }
```

Comment:
this is looping on base actions, not moonbeam, this check should be move above the assignments on lines 649-657